### PR TITLE
TASK: Cleanup NodeTypes package and templates

### PR DESCRIPTION
**Pull request by kitsunet**
Removes the unnecessary ``AssetListImplementation`` and
simplifies NodeType templates.
Additionally sets a dependency to the ``form`` package.

### CODE REVIEW

File: TYPO3.Neos.NodeTypes/composer.json [hunk 1]
```diff
@@ -4,11 +4,12 @@
     "description": "Node type configuration for TYPO3 Neos",
```
Comment (aertmann) [inline TYPO3.Neos.NodeTypes/composer.json]: could remove TYPO3 here

File: TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/Menu.html [hunk 2]
```diff
@@ -1,31 +1,29 @@
...
-		<f:else>
-			<f:security.ifAccess privilegeTarget="TYPO3.Neos:Backend.GeneralAccess">
-				<f:if condition="{node.context.workspace.name} != 'live'">
-					<li>Empty menu</li>
-				</f:if>
-			</f:security.ifAccess>
-		</f:else>
-	</f:if>
+	<f:for each="{items}" as="item">
+		<li{ts:render(path:'{item.state}.attributes', context: {item: item}) -> f:format.raw()}>
```
Comment (aertmann) [inline TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/Menu.html]: missing space after `path:`

File: TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/Partials/SimpleImage.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+		<f:if condition="{neos:rendering.inEditMode()}">
+			<img src="{f:uri.resource(package: 'TYPO3.Neos', path: 'Images/dummy-image.svg')}" title="Dummy image" alt="Dummy image" class="neos-handle" />
+		</f:if>
+	</f:else>
+</f:if>
+
+<f:section name="imageRendering">
+	<media:image asset="{image}" alt="{alternativeText}" title="{title}" width="{width}"
+				 maximumWidth="{maximumWidth}" height="{height}" maximumHeight="{maximumHeight}"
+				 allowUpScaling="{allowUpScaling}" />
```
Comment (aertmann) [inline TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/Partials/SimpleImage.html]: just noticed `allowCropping` is missing here, should we add it?

File: TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/Menu.html [hunk 4]
```diff
@@ -1,31 +1,29 @@
...
-	</ul>
+<f:if condition="{items -> f:count()}">
+	<f:then>
+		<ul>
+			<f:render section="itemsList" arguments="{items: items}" />
+		</ul>
+	</f:then>
+	<f:else>
+		<f:if condition="{neos:rendering.inEditMode()}">
+			<p>Empty menu</p>
```
Comment (aertmann) [inline TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/Menu.html]: should we translate?

File: TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/ContentReferences.html [hunk 5]
```diff
@@ -5,12 +5,10 @@
...
 	<f:else>
-		<f:if condition="{node.context.workspace.name} != 'live'">
-			<f:security.ifAccess privilegeTarget="TYPO3.Neos:Backend.GeneralAccess">
-				<div{attributes -> f:format.raw()}>
-					<p>No references selected</p>
-				</div>
-			</f:security.ifAccess>
+		<f:if condition="{neos:rendering.inEditMode()}">
+			<div{attributes -> f:format.raw()}>
+				<p>No references selected</p>
```
Comment (aertmann) [inline TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/ContentReferences.html]: should we translate?

File: TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/AssetList.html [hunk 6]
```diff
@@ -7,12 +8,10 @@
...
-		<f:security.ifAccess privilegeTarget="TYPO3.Neos:Backend.GeneralAccess">
-			<f:if condition="{node.context.workspace.name} != 'live'">
-				<ul{attributes -> f:format.raw()}>
-					<li>Empty asset list</li>
-				</ul>
-			</f:if>
-		</f:security.ifAccess>
+		<f:if condition="{neos:rendering.inEditMode()}">
+			<ul{attributes -> f:format.raw()}>
+				<li>Empty asset list</li>
```
Comment (aertmann) [inline TYPO3.Neos.NodeTypes/Resources/Private/Templates/NodeTypes/AssetList.html]: should we translate?

**Comment by kitsunet** [comment]
Good catches, will add.

**Comment by kitsunet** [comment]
Should include all of Askes suggestions.

**Comment by aertmann** [comment]
well done 👍

**Comment by johannessteu** [comment]
👍

**Pull request merged**